src/Frame.cpp

```diff
@@ -772,7 +772,7 @@ void Frame::AddImage(int new_width, int new_height, int bytes_per_pixel, QImage:
 	// Create new image object, and fill with pixel data
 	#pragma omp critical (AddImage)
 	{
-		image = std::shared_ptr<QImage>(new QImage(qbuffer, new_width, new_height, new_width * bytes_per_pixel, type, (QImageCleanupFunction) &openshot::Frame::cleanUpBuffer, (void*) qbuffer));
+		image = std::shared_ptr<QImage>(new QImage(qbuffer, new_width, new_height, bytes_per_line, type, (QImageCleanupFunction) &openshot::Frame::cleanUpBuffer, (void*) qbuffer));
```

Comment:
If `openshot::Frame`s are going to contain `QImage`s with possible line-padding, I think this is actually going to necessitate an audit of _all_ of our image-processing code. Because, for instance, I can't see how this **wouldn't**  break this loop:

https://github.com/OpenShot/libopenshot/blob/414a2cdaa3e0b3a009f335670d65179d8366b470/src/effects/Shift.cpp#L79-L84

And potentially dozens more just like it, hiding scattered around the code. We are unfortunately **not** rigorous about using `QImage::[const]scanLine`. Not at _all_.

Reply:
Very good catch. Now, using QImage but NOT using bytesPerLine(), in my opinion, is very poor code hygiene and we should fix all of these cases, since we should not assume only certain types of QImages get here.

However, fixing all of those effects to use bytesPerLine() properly is costly. Here's another option: we could leave the input to AddFrame with this bytes_per_line, but we could copy row-by-row into the QImage _without_ any extra padding, so the QImages remain "packed" and all those effects will work unchanged. This would be slightly lower performance (hundreds of row-by-row copies instead of one bit batch copy), but that may not be meaningful, and it wouldn't change assumptions out from underneath all our users of QImage.

As I said above, I'm open to any/all of these thoughts. Let me know what you think and I'll revise the pull request based on what we decide!